web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/UserSelectionFilter.php

```diff
@@ -324,54 +228,19 @@ protected function getReferencedEntityIdsByBundle(EntityReferenceFieldItemListIn
    *   keyed by entity ID and having the entity labels as values.
    */
   protected function getEntitiesByCategory(): array {
-    $ids = $this->getSparqlQuery()
-      ->graphs(['staging'])
-      ->condition('rid', 'solution')
-      ->execute();
+    $this->ensureEntityDataLoaded();
```

Comment:
Could we just use directly the persistent data store value?

Reply:
We do not store the label of the solution. Should we do so?